src/core/init/duration_updater.ts

```diff
@@ -42,7 +47,11 @@ export default function DurationUpdater(
 ) : Observable<never> {
   return fromEvent(manifest, "manifestUpdate").pipe(
     startWith(null),
-    tap(() => {
+    switchMap(() => {
+      if (mediaSource.readyState === "ended") {
+        return EMPTY;
```

Comment:
Why not also postponing when it is equal to "ended"?
Here if we perform `endOfStream` then find out new segments are available (and find out that a new duration is also available), we risk to never update the MediaSource's duration, no?

That also brings the question: is a `sourceopen` event sent when MediaSource goes from `ended` to `open`?

Reply:
You're right !